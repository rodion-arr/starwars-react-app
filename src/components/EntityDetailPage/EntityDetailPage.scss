.entity-detail {
  padding-top: 5px;

  &__title-wrap {
    margin: 10px 0;
    padding: 0 var(--global-padding);
  }

  &__title {
    padding: 5px 0;
    font-weight: bold;
    text-transform: uppercase;
    border-top: solid var(--main-color) 1px;
    border-bottom: solid var(--main-color) 1px;
  }

  &__img-wrap {
    margin: 0 var(--global-padding);
    background-color: rgba(255, 255, 255, 0.5);
    border-radius: var(--global-border-radius);
    padding: 10px;
  }

  &__img {
    display: block;
    margin: 0 auto;
    max-height: 500px;
    max-width: 100%;
  }

  &__spec {
    margin: 0 var(--global-padding);
  }
}
