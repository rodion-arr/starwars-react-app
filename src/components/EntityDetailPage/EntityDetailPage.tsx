import './EntityDetailPage.scss';
import {CharacteristicsTable} from "../CharacteristicsTable/CharacteristicsTable";
import React from "react";

interface Props {
  name: string;
  image: string;
  charProps?: Record<string, string | JSX.Element>
  children? : JSX.Element;
}

export const EntityDetailPage = ({
  name,
  image,
  charProps,
  children,
}: Props): JSX.Element => {
  return (
    <div className="entity-detail">
      <div className="entity-detail__title"><h2>{name}</h2></div>
      <div className="entity-detail__img"><img src={image} alt={name}/></div>
      {charProps && (<>
        <div className="entity-detail__title"><h2>Characteristics</h2></div>
        <CharacteristicsTable rows={charProps} className="entity-detail__spec" />
      </>)}
      {children}
    </div>
  );
};
