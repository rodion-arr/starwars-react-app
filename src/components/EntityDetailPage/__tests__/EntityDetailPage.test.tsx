import {render} from "@testing-library/react";
import {EntityDetailPage} from "../EntityDetailPage";

describe('EntityDetailPage', () => {
  it('should render correctly', function () {
    const {container} = render(<EntityDetailPage
      name="Unit test"
      image="/test"
      charProps={{
        test: 'value',
        test2: 'value 2',
      }}
    >
      <span>children</span>
    </EntityDetailPage>);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly without optional params', function () {
    const {container} = render(<EntityDetailPage
      name="Unit test"
      image="/test"
    />);

    expect(container).toMatchSnapshot();
  });
});
