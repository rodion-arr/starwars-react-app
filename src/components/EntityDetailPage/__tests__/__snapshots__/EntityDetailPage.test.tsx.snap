// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityDetailPage should render correctly 1`] = `
<div>
  <div
    class="entity-detail"
  >
    <div
      class="entity-detail__title-wrap"
    >
      <h2
        class="entity-detail__title"
      >
        Unit test
      </h2>
    </div>
    <div
      class="entity-detail__img-wrap"
    >
      <img
        alt="Unit test"
        class="entity-detail__img"
        src="/test"
      />
    </div>
    <div
      class="entity-detail__title-wrap"
    >
      <h2
        class="entity-detail__title"
      >
        Characteristics
      </h2>
    </div>
    <div
      class="entity-detail__spec detail-spec-table"
    >
      <table
        class="detail-spec-table__table"
      >
        <tbody>
          <tr>
            <td>
              test
            </td>
            <td>
              value
            </td>
          </tr>
          <tr>
            <td>
              test2
            </td>
            <td>
              value 2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <span>
      children
    </span>
  </div>
</div>
`;

exports[`EntityDetailPage should render correctly without optional params 1`] = `
<div>
  <div
    class="entity-detail"
  >
    <div
      class="entity-detail__title-wrap"
    >
      <h2
        class="entity-detail__title"
      >
        Unit test
      </h2>
    </div>
    <div
      class="entity-detail__img-wrap"
    >
      <img
        alt="Unit test"
        class="entity-detail__img"
        src="/test"
      />
    </div>
  </div>
</div>
`;
